public_html/templates/header.html

```diff
@@ -1,16 +1,60 @@
-	{% if user %}
-		<a href="{{ urlFor( 'logout' ) }}" id="login-btn" class="btn btn-sm btn-default {{ user ? 'logged-in' : '' }}">
-			<span class=" glyphicon glyphicon-log-in"></span> {{ 'logout' | message }}<br>
-			<small>{{ 'header-loggedin' | message( user.name ) }}</small>
-		</a>
-	{% else %}
-		<a href="{{ urlFor( 'oauth_init' ) }}" id="login-btn" class="btn btn-sm btn-default">
-			<span class=" glyphicon glyphicon-log-in"></span> {{ 'login' | message }}
-		</a>
-	{% endif %}
+	<nav class="navbar navbar-default">
+		<div class="container-fluid">
+			<!-- Brand and toggle get grouped for better mobile display -->
+			<div class="navbar-header">
+				<button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1" aria-expanded="false">
+					<span class="sr-only">Toggle navigation</span>
+					<span class="icon-bar"></span>
+					<span class="icon-bar"></span>
+					<span class="icon-bar"></span>
+				</button>
+				<a class="navbar-brand" href="{{ urlFor( 'home', {wikiLang:wikiLang} ) }}">
+					<img src="{{ siteUrl( 'images/copypatrol.png' ) }}" />
+				</a>
+			</div>
+
+			<!-- Collect the nav links, forms, and other content for toggling -->
+			<div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
+				<ul class="nav navbar-nav">
+					<li class="dropdown">
+						<a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-haspopup="true" aria-expanded="false">
+							{{ (wikiLang~'-wikipedia') | message }} <span class="caret"></span>
+						</a>
+						<ul class="dropdown-menu">
+							{% if wikiLang != 'en' %}
+							<li><a href="{{ urlFor( currentRoute, {wikiLang:'en'} ) }}">{{ ('en-wikipedia') | message }}</a></li>
+							{% endif %}
+							{% if wikiLang != 'fr' %}
+							<li><a href="{{ urlFor( currentRoute, {wikiLang:'fr'} ) }}">{{ ('fr-wikipedia') | message }}</a></li>
+							{% endif %}
+						</ul>
+					</li>
+					<li><a target="_blank" href="https://meta.wikimedia.org/wiki/CopyPatrol">{{ 'documentation' | message }}</a></li>
+					<li><a target="_blank" href="https://meta.wikimedia.org/wiki/Talk:CopyPatrol">{{ 'feedback' | message }}</a></li>
+					<li><a href="{{ urlFor( 'leaderboard', {wikiLang:wikiLang} ) }}">{{ 'leaderboard' | message }}</a></li>
+				</ul>
+				<ul class="nav navbar-nav navbar-right">
+
+					{% if user %}
+					<li><span class="navbar-text">{{ 'header-loggedin' | message( user.name ) }}</span></li>
+					<li>
+						<form action="{{ urlFor( 'logout' ) }}" method="get">
+							<button class="btn btn-default navbar-btn logged-in">
+								<span class="glyphicon glyphicon-log-out"></span> {{ 'logout' | message }}
+							</button>
+						</form>
+					</li>
+					{% else %}
+					<form action="{{ urlFor( 'oauth_init' ) }}" method="get">
+						<button class="btn btn-default navbar-btn">
+							<span class="glyphicon glyphicon-log-in"></span> {{ 'login' | message }}
+						</button>
+					</form>
```

Comment:
Maybe make the `navbar-nav` a `<div>` or `<span>`, then under `{% if user %}` add the `<ul>` and `<li>`s, since `<form>` should not be a child of `<ul>`